lib/ansible/module_utils/vmware.py

```diff
@@ -799,6 +801,20 @@ def set_vm_power_state(content, vm, state, force):
     return result
 
 
+def wait_for_poweroff(vm, max_poll=300, interval=1):
+    result = dict()
+    poll = 0
+    while poll <= max_poll:
+        if vm.runtime.powerState.lower() == 'poweredoff':
+            break
+        time.sleep(interval)
+        poll += 1
+    else:
+        result['failed'] = True
+        result['msg'] = 'Timeout while waiting for VM power off.'
+    return result
```

Comment:
interval=1 is too short, I think it will be too eavy on the API for big deployment.
The recommended value seems to be 15s : https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/vmware.py#L56

Also I am not very fan of the fact you count the polls instead of the time. Personally I would have prefer a `max_waiting_time` instead of a `max_poll`.

Currently the real max_poll is actualy max_poll + 1 since you use `poll <= max_pool`.

Reply:
Agreed